packages/@lwc/template-compiler/src/parser/attribute.ts

```diff
@@ -253,16 +249,11 @@ function isTemplateDirective(attrName: string): boolean {
     });
 }
 
-function shouldCamelCaseAttribute(element: IRElement, attrName: string) {
-    const { tag } = element;
-    const isDataAttributeOrFmk = isDataAttribute(attrName) || isFmkAttribute(attrName);
-    const isSvgTag = SUPPORTED_SVG_TAGS.has(tag);
-    return !isSvgTag && !isDataAttributeOrFmk;
-}
-
-export function attributeToPropertyName(element: IRElement, attrName: string): string {
-    if (!shouldCamelCaseAttribute) {
-        return attrName;
-    }
-    return toPropertyName(ATTRS_PROPS_TRANFORMS[attrName] || attrName);
+/**
+ * convert attribute name from kebab case to camel case property name
+ * Note: Should not invoke this function for data attribute
```

Comment:
Why not?

Reply:
data attributes are not processed as properties. They are always set as attributes.

Reply:
I will remove the comment. Its an internal detail of the package.